cores/esp8266/StreamString.h

```diff
@@ -1,39 +1,292 @@
 /**
- StreamString.h
+    StreamString.h
 
- Copyright (c) 2015 Markus Sattler. All rights reserved.
- This file is part of the esp8266 core for Arduino environment.
+    Copyright (c) 2020 D. Gauchard. All rights reserved.
+    This file is part of the esp8266 core for Arduino environment.
 
- This library is free software; you can redistribute it and/or
- modify it under the terms of the GNU Lesser General Public
- License as published by the Free Software Foundation; either
- version 2.1 of the License, or (at your option) any later version.
+    This library is free software; you can redistribute it and/or
+    modify it under the terms of the GNU Lesser General Public
+    License as published by the Free Software Foundation; either
+    version 2.1 of the License, or (at your option) any later version.
 
- This library is distributed in the hope that it will be useful,
- but WITHOUT ANY WARRANTY; without even the implied warranty of
- MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
- Lesser General Public License for more details.
+    This library is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+    Lesser General Public License for more details.
 
- You should have received a copy of the GNU Lesser General Public
- License along with this library; if not, write to the Free Software
- Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
+    You should have received a copy of the GNU Lesser General Public
+    License along with this library; if not, write to the Free Software
+    Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
 
 */
 
-#ifndef STREAMSTRING_H_
-#define STREAMSTRING_H_
+#ifndef __STREAMSTRING_H
+#define __STREAMSTRING_H
 
+#include "WString.h"
 
-class StreamString: public Stream, public String {
+///////////////////////////////////////////////////////////////
+// S2Stream points to a String and makes it a Stream
+// (it is also the helper for StreamString)
+
+class S2Stream: public Stream
```

Comment:
I read in the readme that this is meant to stream a String without making a copy of the string. That is not evident from the code. If that is the purpose of having this class be separate, then I suggest changing the name to reflect that, e.g.: StringViewStream or StringReferenceStream or something similar.

Reply:
> That is not evident from the code

Can you elaborate ?

> If that is the purpose of having this class be separate

The purpose is to indifferently use Strings where Streams can be used.
`StringReferenceStream` is the right meaning but is a hell of a name to type.
I'm open to other suggestions :)